app/uk/gov/hmrc/tai/service/TaxCodeChangeService.scala

```diff
@@ -116,17 +116,49 @@ class TaxCodeChangeServiceImpl @Inject()(taxCodeChangeConnector: TaxCodeChangeCo
     }
   }
 
-  private def addEndDate(date: LocalDate, taxCodeRecord: TaxCodeRecord): TaxCodeRecordWithEndDate ={
+  private def addEndDate(date: LocalDate, taxCodeRecord: TaxCodeRecord): TaxCodeRecordWithEndDate = {
     TaxCodeRecordWithEndDate(
       taxCodeRecord.taxCode, taxCodeRecord.basisOfOperation, taxCodeRecord.dateOfCalculation, date,
       taxCodeRecord.employerName, taxCodeRecord.payrollNumber, taxCodeRecord.pensionIndicator, taxCodeRecord.isPrimary
     )
   }
 
+  def latestTaxCodes(nino:Nino, taxYear: TaxYear)(implicit hc: HeaderCarrier):Future[Seq[TaxCodeRecordWithEndDate]] = {
+      mostRecentTaxCodeRecordInYear(nino, taxYear)
+  }
+
   private def taxCodeHistory(nino: Nino, taxYear: TaxYear): Future[TaxCodeHistory] = {
     taxCodeChangeConnector.taxCodeHistory(nino, taxYear, taxYear.next)
   }
 
+  def mostRecentTaxCodeRecordInYear(nino: Nino, year: TaxYear): Future[Seq[TaxCodeRecordWithEndDate]] = {
+
+    taxCodeChangeConnector.taxCodeHistory(nino, year, year.next).map { taxCodeHistory =>
+
+      val groupedTaxCodeRecords: Map[String, Seq[TaxCodeRecord]] = taxCodeHistory.taxCodeRecords.groupBy(_.employerName)
+
+      val thing = groupedTaxCodeRecords.values.flatMap {
+        taxCodeRecords =>
+          val sorted = taxCodeRecords.sortBy(_.dateOfCalculation)
```

Comment:
There was a utility method created for this on https://github.com/hmrc/tai/pull/61/files#diff-2b8e75b096d3d89321bd270ca8f70e0fR40